_posts/2024-12-20-building-databases-for-the-right-user.md

```diff
@@ -0,0 +1,72 @@
+---
```

Comment:
Based on feedback from a first reader (not sure if they want to be identified)
* I wonder if, given the motivating question isn't doing me favors in the post, I can even remove it without harming the message
* Part of not touching privacy is that I don't believe you can get it as long as it's someone else's app. This is why I go for sovereignty, but I should call that out and clarify.
* Relatedly, there's the question of what "your" data is. Perhaps a better framing is "data you share with an application to get value from the application, with the hope that you'll be able to use the data for whatever you'd like in the future." For example: your run data in Strava is data you shared with Strava to track your runs, and you expect to be able to slice an dice it however you like. Whatever click tracking Strava does to later analyze their funnel is not data you shared with them, and (even if you don't appreciate they tracked you) not data you were hoping to later have access to.
* Is key management on the critical path to "owning your data?" Not in a way the user has to think about it. The system would be managing this for you (e.g., in `ayb`, I can `ayb client share marcua/test.sqlite some-user read-write`, and as the feedback I got suggests, you could introduce OAuth2 for a more traditional way to auth your database to their app). That said, maybe that part of the dance is complicated. I could cut it out without loss of clarity --- credentials are obviously part of "here is my database," but it's probably too technical for this writeup.
* Maybe Terms of Data Use should get its own post with a bit more comparison to related work